app/models/manageiq/providers/inventory/persister/builder.rb

```diff
@@ -208,26 +211,18 @@ def merge_hashes(source, dest, mode)
       def auto_model_class
         model_class = begin
           # a) Provider specific class
-          provider_module = ManageIQ::Providers::Inflector.provider_module(@persister_class).name
-          manager_module = self.class.name.split('::').last
```

Comment:
We were trying to guess the ems class here when we could just initialize the builder with it, this also allows us to use the manager_class when building the persisters so e.g. we can build model_class for subclasses that don't match their root name e.g. MiqTemplate is usually ManageIQ::Providers::Vendor::InfraManager::Template and EmsCluster is usually `ManageIQ::Providers::Vendor::InfraManager::`